.circleci/run_xpu_tests.py

```diff
@@ -79,10 +79,14 @@
 
         patch_sklearn()
 
-    if args.device == "gpu":
-        from sklearnex._config import config_context
+    from sklearnex._config import config_context
 
-        with config_context(target_offload=args.device, allow_fallback_to_host=False):
+    with config_context(allow_sklearn_after_onedal=False):
+
+        if args.device == "gpu":
+            with config_context(target_offload=args.device, allow_fallback_to_host=False):
```

Comment:
Nested `config_context` doesn't stack - lower context overwrites upper, see `allow_sklearn_after_onedal`:
```python
>>> from sklearnex import config_context, get_config
>>> with config_context(allow_sklearn_after_onedal=False):
...   with config_context(target_offload="gpu", allow_fallback_to_host=False):
...     c = get_config()
...     print(c)
...
```
`{'assume_finite': False, 'working_memory': 1024, 'print_changed_only': True, 'display': 'diagram', 'pairwise_dist_chunk_size': 256, 'enable_cython_pairwise_dist': True, 'array_api_dispatch': False, 'transform_output': 'default', 'enable_metadata_routing': False, 'skip_parameter_validation': False, 'target_offload': 'gpu', 'allow_fallback_to_host': False, 'allow_sklearn_after_onedal': True}`


Is it expected?

Reply:
Very good catch.
It is expected that lower context overwrites upper, but only in case if it explicitly updated on the lower context with config_context param. The logic was broken for `allow_sklearn_after_onedal`, update it. Will cover this case with test for nested config_context check for our scikit-learn-intelex config params

Reply:
Fixed at https://github.com/intel/scikit-learn-intelex/pull/1907/commits/50d569fd642a463a45b44bee297bff037e3e3aad and covered with new tests see https://github.com/intel/scikit-learn-intelex/pull/1907/commits/205289e8758b2cc0f4e1364b348c4f36d6484f99